README.md

```diff
@@ -1,103 +1,124 @@
 # C++ example module
 
-This is our example module for C++. It shows you how to:
+This example module demonstrates the use of the [C++ SDK](https://github.com/viamrobotics/viam-cpp-sdk) to build a minimal Viam module.
+It is [available for use on the registry](https://app.viam.com/module/viam/module-example-cpp) and shows how to include CI (Continuous Integration) builds for both x86 and ARM architectures.
 
-- Build a minimal Viam module using our [C++ SDK](https://github.com/viamrobotics/viam-cpp-sdk)
-- With CI builds for x86 and ARM
-- Released through our registry
+You can use this as a reference along with the C++ instructions for [coding your own modules](https://docs.viam.com/registry/create/).
 
-## Local build environment
 
-Known good environments:
-- Debian Bullseye (this is what our CI uses)
-- You can use Ubuntu 22.04 for testing, but the C++ builds will not be binary compatible with Bullseye
+## Requirements
 
-Setup and build instructions:
-- Install the apt packages in [apt-setup.sh](apt-setup.sh)
-- Install cmake
-	- Check CMAKE_VERSION in the github workflow (3.27.4 as of this writing)
-	- On Ubuntu 22.04, `snap install cmake` is new enough but `apt install cmake` is not
-	- Otherwise follow [CMake's download instructions](https://cmake.org/download/)
-- [Install rust](https://www.rust-lang.org/tools/install) (needed for viam_rust_utils)
+Before building this module, verify your operating system's development environment is compatible.
+The tested ideal environment for building this module is Debian Bullseye (used in our CI).
 
-## Deploy to robot
+You can also use Ubuntu 22.04 for testing, but the resulting C++ builds will not be binary compatible with Bullseye.
 
-These instructions assume you're targeting something similar to Debian bullseye.
 
-First install libraries on your target system:
+To setup for building this module:
+
+1. Install the apt packages in [apt-setup.sh](apt-setup.sh).
+2. Install CMake. Ensure your version of CMake matches the [CMAKE_VERSION in this repository's github workflow](https://github.com/viamrobotics/module-example-cpp/blob/af057c9169a730b2c5de5c26f877c0419fb52c19/.github/workflows/build.yml#L7).
+	- To install on Ubuntu 22.04, run `snap install cmake` or follow [CMake's download instructions](https://cmake.org/download/).
+3. [Install rust](https://www.rust-lang.org/tools/install) (needed for `viam_rust_utils`).
+
+## Build and Run
```

Comment:
'build and run' here is probably copying my old instructions, which were low-key wrong

running from registry + running from locally built module are different cases (first one is easy, second one is hard)

imo we should guide people through these two steps for example code (install from reg to see it work, then build), but 1) this is just an opinion, and 2) would require a bit of testing to make sure it's actually doable. (local exec will likely collide with installed one)

Reply:
Addressing this in the next commit.